pkg/controller/populators/populator-base.go

```diff
@@ -358,6 +358,10 @@ func (r *ReconcilerBase) reconcileCommon(pvc *corev1.PersistentVolumeClaim, popu
 
 func (r *ReconcilerBase) reconcileCleanup(pvcPrime *corev1.PersistentVolumeClaim) (reconcile.Result, error) {
 	if pvcPrime != nil {
```

Comment:
not related to this pr but you could really improve the indentation in this func if you return early when `pvcPrime == nil`

Reply:
Wanted to avoid redundant `returns` but I agree, I'll change it.